src/main/java/games/strategy/util/Util.java

```diff
@@ -122,4 +130,19 @@ public static String getStringFromInputStream(InputStream in) {
     }
     return builder.toString();
   }
+
+  /**
+   * Creates a hash of the given String based on the SHA-512 algorithm.
+   * 
+   * @param input The input String to hash.
+   * @return A hashed hexadecimal String of the input.
+   */
+  public static String sha512(String input) {
```

Comment:
I think more the concern is unit tests for anything that uses this. If under the covers a client uses a sha512, the unit test would likely also need to know about that and do the same to create an expected value. That means any client using this code is coupled to it and the unit test. My suggestion is more to see if anything using a sha512 can be encapsulated into a single module, then no test other than the one test checking that single module need know about sha512. All other tests would instead mock the sha512 module and then fake the results explicitly.